### PR TITLE
UI Localization

### PR DESCRIPTION
**Pull request by snixtho**
- Adds a way for modules to register localization to be used anywhere.
- Can adapt to a player's own selected language.
- Ability to use indexer and dynamic object accessors, depending on the situation that fits best. (Dynamic is useful for Manialinks)
- Can translate locale names within arbitrary strings. Useful for setting up localization for compile-time constants such as command descriptions.
- Can configure a default language for the controller.

All current modules have their UI strings replaced with the localization system (hence the amount of files changed).

All localization and translations are managed here: https://crowdin.com/project/evosc-sharp

### CODE REVIEW

**Review by AtomicLiquid — CHANGES_REQUESTED** [review]

File: src/Modules/MapsModule/Controllers/MapsController.cs [hunk 1]
```diff
@@ -73,9 +76,9 @@ public async Task RemoveMap(long mapId)
         Context.AuditEvent.Success()
             .WithEventName(AuditEvents.MapRemoved)
             .HavingProperties(new {Map = map})
-            .Comment("Map was removed.");
+            .Comment(_locale.Audit_MapRemoved);
         
-        await _server.SuccessMessageAsync($"Removed map with ID {mapId} from the maplist.");
+        await _server.SuccessMessageAsync(_locale.PlayerLanguage.MapRemovedSuccessfully(mapId), Context.Player);
```
[the remark this answers is not in the document]
Comment (snixtho) [reply src/Modules/MapsModule/Controllers/MapsController.cs:81]: Can you elaborate?

[the remark this answers is not in the document]
Comment (AtomicLiquid) [reply src/Modules/MapsModule/Controllers/MapsController.cs:81]: image
MapAddedSuccessfully and MapRemovedSuccessfully should have the same structure.

**Review by kdankert — COMMENTED** [review]
Looks fine for me. We should definitely find a better way to mock contexts inside controllers. Also it would be great if test coverage could be improved. We should also integration test migrations to ensure they are working together (unit tests do not cover cases that might appear).

**Comment by snixtho** [comment]
> Looks fine for me. We should definitely find a better way to mock contexts inside controllers. Also it would be great if test coverage could be improved. We should also integration test migrations to ensure they are working together (unit tests do not cover cases that might appear).

I agree, the tests in this PR are mostly only for what was actually added. There are a bunch of services and things in the core that needs tests, which we should create soon. I could make some issues on that.

**Review by AtomicLiquid — APPROVED** [review]

**Pull request merged**